packages/browser-agent-core/features/ajax/instrument/index.js

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { originals, getLoaderConfig, getRuntime } from '../../../common/config/config'
+import { handle } from '../../../common/event-emitter/handle'
+import { id } from '../../../common/ids/id'
+import { ffVersion } from '../../../common/browser-version/firefox-version'
+import { dataSize } from '../../../common/util/data-size'
+import { eventListenerOpts } from '../../../common/event-listener/event-listener-opts'
+import { now } from '../../../common/timing/now'
+import { wrapFetch, wrapXhr } from '../../../common/wrap'
+import { parseUrl } from '../../../common/url/parse-url'
+import { DT } from './distributed-tracing'
+import { responseSizeFromXhr } from './response-size'
+import { FeatureBase } from '../../../common/util/feature-base'
+
+var handlers = ['load', 'error', 'abort', 'timeout']
+var handlersLen = handlers.length
+
+var origRequest = originals.REQ
+var origXHR = window.XMLHttpRequest
+
+export class Instrument extends FeatureBase {
+  constructor(agentIdentifier) {
+    super(agentIdentifier)
+    const agentRuntime = getRuntime(this.agentIdentifier);
+    // // NREUM.debug("initialize ajax instrument!", agentIdentifier)
+
+    // Don't instrument Chrome for iOS, it is buggy and acts like there are URL verification issues
+    if (!agentRuntime.xhrWrappable || agentRuntime.disabled) return
+
+    agentRuntime.features.xhr = true;   // declare that we are using xhr instrumentation
+
+    this.dt = new DT(this.agentIdentifier)
+
+    this.handler = (type, args, ctx, group) => handle(type, args, ctx, group, this.ee)
+    this.wrappedFetch = getWrappedFetch(this.ee)
```

Comment:
I don't see this used anywhere in the agent. Is this from the original modularity prototype? Can we delete it?